File: behdavar-common/src/main/java/com/bar/behdavarcommon/enumeration/PhoneType.java
package com.bar.behdavarcommon.enumeration;

public enum PhoneType implements AbstractEnum<String> {
    PHONE("P"),
    MOBILE("M"),
    FAX("F");

    private String value;
    ;

    PhoneType(String s) {
        this.value = s;
    }

    @Override
    public String getValue() {
        return this.value;
    }

//    public static class PhoneTypeConverter extends AbstractEnumConverter<PhoneType, String> {
//        public PhoneTypeConverter(Class<PhoneType> clazz) {
//            super(clazz);
//        }
//    }
}

File: behdavar-front/src/app/model/enum/PhoneType.ts
import {EnumValueTitle} from "./EnumValueTitle";
import {PhoneTypeLang} from "../lang";

const phoneTypeLang: PhoneTypeLang = new PhoneTypeLang();

export enum PhoneType {
  PHONE = "PHONE",
  MOBILE = "MOBILE",
  FAX = "FAX"
}


export const PHONE_TYPE_TITLE: EnumValueTitle<PhoneType>[] = [
  {value: PhoneType.PHONE, title: phoneTypeLang.phone},
  {value: PhoneType.MOBILE, title: phoneTypeLang.mobile},
  {value: PhoneType.FAX, title: phoneTypeLang.fax},

];
